components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1836,73 +1836,93 @@ public IdentityProvider getEnabledIdPByRealmId(String realmId, String tenantDoma
+                                            }
+                                        }
+                                        StringBuilder certificate = new StringBuilder();
                                         try {
-                                            FederatedAuthenticatorConfig metaFederated = metadataConverter.getFederatedAuthenticatorConfig(properties, certificate);
+                                            FederatedAuthenticatorConfig metaFederated = metadataConverter
+                                                    .getFederatedAuthenticatorConfig(properties, certificate);
 
                                             String spName = "";
 
-                                            for (int b = 0; b < properties.length; b++) {
-                                                if (properties[b] != null && properties[b].getName() != null &&
-                                                        properties[b].getName().toString().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)) {
-                                                    spName = properties[b].getValue();
+                                            for (Property value : properties) {
+                                                if (value != null && value.getName() != null &&
+                                                        value.getName()
+                                                                .equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)) {
+                                                    spName = value.getValue();
                                                 }
                                             }
                                             if (spName.equals("")) {
                                                 throw new IdentityProviderManagementException("SP name can't be empty");
                                             }
 
-                                            if (metaFederated != null && ArrayUtils.isNotEmpty(metaFederated.getProperties())) {
+                                            if (metaFederated != null &&
+                                                    ArrayUtils.isNotEmpty(metaFederated.getProperties())) {
                                                 for (int y = 0; y < metaFederated.getProperties().length; y++) {
-                                                    if (metaFederated.getProperties()[y] != null && metaFederated.getProperties()[y].getName() != null
-                                                            && metaFederated.getProperties()[y].getName().toString().equals(IdentityApplicationConstants.Authenticator.SAML2SSO.SP_ENTITY_ID)) {
+                                                    if (metaFederated.getProperties()[y] != null &&
+                                                            metaFederated.getProperties()[y].getName() != null
```

Comment:
Shall we invert the equals method to have the constant first so that we can omit the null check? WDYT?